src/Header/RequestHandler.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef GROUP4_HEADER_REQUESTHANDLER_H_
+#define GROUP4_HEADER_REQUESTHANDLER_H_
+
+#include "OperateData.h"
+#include "Authenticator.h"
+#include "APIHandler.h"
+#include <string>
+#include <unordered_map>
+#include <memory>
+
+//Will decode the request string and check if it's a valid request.
+//calling RequestParser to get OperateData then call correspond api function
+class RequestHandler
+{
+public:
+    //instantiate instance with request, will parse to OperateData and check with Authenticator and call correspond api.
+    RequestHandler(const byte &requestData);
```

Comment:
If you want to take a socket:
- If you're using asio, you'll take this: https://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/ip__tcp/socket.html
- If you are using raw unix socket, you'll take a file descriptor (int).

Reply:
request will be string from file descriptor